pkg/trace/writer/stats_test.go

```diff
@@ -133,6 +142,10 @@ func TestStatsWriter(t *testing.T) {
 			assert.Equal(1, len(payloads[i].Stats))
 			assert.Equal(1, len(payloads[i].Stats[0].Stats))
 			assert.Equal(expectedNbEntriesByPayload[i], len(payloads[i].Stats[0].Stats[0].Stats))
+			previous := payloads[i].Stats[0].Stats
+			payloads[i].Stats[0].Stats = nil
+			assert.Equal(baseClientPayload, payloads[i].Stats[0])
+			payloads[i].Stats[0].Stats = previous
```

Comment:
Not sure I understand why these steps are needed?

Reply:
Yeah, I did it a weird way, let me update it to make it easier

Reply:
Updated.